data-prepper-plugins/csv-processor/src/main/java/com/amazon/dataprepper/plugins/processor/csv/CSVProcessor.java

```diff
@@ -45,4 +83,65 @@ public boolean isReadyForShutdown() {
     public void shutdown() {
 
     }
+
+    private CsvMapper createCsvMapper() {
+        final CsvMapper mapper = new CsvMapper();
+        mapper.enable(CsvParser.Feature.WRAP_AS_ARRAY); // allows mapper to read with empty schema
+        return mapper;
+    }
+
+    private CsvSchema createCsvSchema() {
+        final char delimiterAsChar = config.getDelimiter().charAt(0); // safe due to config input validations
+        final char quoteCharAsChar = config.getQuoteCharacter().charAt(0); // safe due to config input validations
+        final CsvSchema schema = CsvSchema.emptySchema().withColumnSeparator(delimiterAsChar).withQuoteChar(quoteCharAsChar);
+        return schema;
+    }
+
+    private List<String> parseHeader(Event event, final boolean thisEventHasHeaderSource, final CsvMapper mapper,
+                                     final CsvSchema schema) {
+        if (thisEventHasHeaderSource) {
+            try {
+                final String headerUnprocessed = event.get(config.getColumnNamesSourceKey(), String.class);
+                final MappingIterator<List<String>> headerIterator = mapper.readerFor(List.class).with(schema)
+                        .readValues(headerUnprocessed);
+                // if header is empty, behaves correctly since columns are autogenerated.
+                final List<String> headerFromEventSource = headerIterator.nextValue();
+                return headerFromEventSource;
+            }
+            catch (IOException e) {
+                final List<String> emptyHeader = new ArrayList<String>();
+                return emptyHeader;
+            }
```

Comment:
Not important but you might consider moving this into its own function as well. I usually try to avoid having a `try/catch` block within an `if` if there are multiple branches already.